win32/Lib/pywin32_testutil.py

```diff
@@ -229,42 +225,28 @@ def addError(self, test, err):
             and exc_val.winerror in non_admin_error_codes
             and not check_is_admin()
         ):
-            exc_val = TestSkipped(exc_val)
+            return self.addSkip(test, str(exc_val))
         # and COM errors due to objects not being registered (the com test
         # suite will attempt to catch this and handle it itself if the user
         # is admin)
-        elif isinstance(exc_val, pywintypes.com_error) and exc_val.hresult in [
+        elif isinstance(exc_val, pywintypes.com_error) and exc_val.hresult in {
             winerror.CO_E_CLASSSTRING,
             winerror.REGDB_E_CLASSNOTREG,
             winerror.TYPE_E_LIBNOTREGISTERED,
-        ]:
-            exc_val = TestSkipped(exc_val)
-        # NotImplemented generally means the platform doesn't support the
-        # functionality.
+        }:
+            return self.addSkip(test, str(exc_val))
+        # NotImplemented generally means the platform doesn't support the functionality.
         elif isinstance(exc_val, NotImplementedError):
-            exc_val = TestSkipped(NotImplementedError)
-
-        if isinstance(exc_val, TestSkipped):
-            reason = exc_val.args[0]
-            # if the reason itself is another exception, get its args.
-            try:
-                reason = tuple(reason.args)
-            except (AttributeError, TypeError):
-                pass
-            self.skips.setdefault(reason, 0)
-            self.skips[reason] += 1
-            if self.showAll:
-                self.stream.writeln(f"SKIP ({reason})")
-            elif self.dots:
-                self.stream.write("S")
-                self.stream.flush()
-            return
+            return self.addSkip(test, str(exc_val))
+
         super().addError(test, err)
 
-    def printErrors(self):
+    def printErrors(self) -> None:
         super().printErrors()
-        for reason, num_skipped in self.skips.items():
-            self.stream.writeln("SKIPPED: %d tests - %s" % (num_skipped, reason))
+        reasons = [reason for (test, reason) in self.skipped]
+        reason_counts = [(reason, reasons.count(reason)) for reason in set(reasons)]
+        for reason, num_skipped in reason_counts:
+            self.stream.writeln(f"SKIPPED: {num_skipped} tests - {reason}")
```

Comment:
There's probably a more efficient way to write this. Maybe using `itertools.groupby`?

Reply:
Found it: https://docs.python.org/3/library/collections.html#collections.Counter